### PR TITLE
AdvancedBallistics

### PR DESCRIPTION
**Pull request by ulteq**
* Once the AdvancedBallistics.dll is compiled we can use handleFiredExtension instead of handleFired
* The winddeflection module should be disabled automatically when the advanced_ballistics module is used
* Do we want to keep the intuitive wind info (shift +k) inside of the advanced_ballistics module or rather move it into the weather module?
* Altitude and Latitude are currently set inside of XEH_postInit. This should also better be done outside of advanced_ballistics

Edit: Everything above is solved.

### CODE REVIEW

File: addons/advanced_ballistics/functions/fnc_calculateRoughnessLength.sqf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// Source: http://es.ucsc.edu/~jnoble/wind/extrap/index.html
+_roughness_lengths = [0.0002, 0.0005, 0.0024, 0.03, 0.055, 0.1, 0.2, 0.4, 0.8, 1.6];
+
+_windSource = _this vectorDiff ((vectorNormalized wind) vectorMultiply 25);
+
+_nearBuildings = count (_windSource nearObjects ["Building", 50]);
+_isWater = surfaceIsWater _windSource;
+
+if (_nearBuildings == 0 && _isWater) exitWith {
```
Comment (Glowbal) [inline addons/advanced_ballistics/functions/fnc_calculateRoughnessLength.sqf]: Above two lines could be replaced with:

```if (_nearBuildings && surfaceIsWater _windSource) exitwith {```

Removes one variable assignment that is only used in this spot.

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+    if ((GVAR(bulletDatabaseOccupiedIndices) pushBack _index) == 0) then {
+        [{
+            private ["_bulletDatabaseEntry", "_index", "_bullet", "_caliber", "_bulletTraceVisible", "_bulletVelocity", "_bulletPosition"];
+
+            {
+                _bulletDatabaseEntry = (GVAR(bulletDatabase) select _x);
+                _bullet = _bulletDatabaseEntry select 0;
+                _index  = _bulletDatabaseEntry select 3;
+                if (!alive _bullet) then {
```
Comment (Glowbal) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: `_index` is only used after this if statement. The assignement should be after this if statement, as that is when we need it. Assignment of the bullet variable is also not necessary here.

```
if (!alive (_bulletDatabaseEntry select 0) /* bullet */) then {
   _index = _bulletDatabaseEntry select 3;
...
```

File: addons/advanced_ballistics/functions/fnc_calculateRoughnessLength.sqf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Public: No
+ */
+#include "script_component.hpp"
+
+private ["_roughness_lengths", "_windSource", "_nearBuildings", "_isWater"];
+
+// Source: http://es.ucsc.edu/~jnoble/wind/extrap/index.html
+_roughness_lengths = [0.0002, 0.0005, 0.0024, 0.03, 0.055, 0.1, 0.2, 0.4, 0.8, 1.6];
+
+_windSource = _this vectorDiff ((vectorNormalized wind) vectorMultiply 25);
```
Comment (commy2) [inline addons/advanced_ballistics/functions/fnc_calculateRoughnessLength.sqf]: ACE_wind ?

**Comment by ulteq** [comment]
Ready to be merged if there are no objections.

**Pull request merged**